core/blockchain.go

```diff
@@ -947,10 +947,18 @@ func (bc *BlockChain) Stop() {
 	//  - HEAD:     So we don't need to reprocess any blocks in the general case
 	//  - HEAD-1:   So we don't do large reorgs if our HEAD becomes an uncle
 	//  - HEAD-127: So we have a hard limit on the number of blocks reexecuted
+	//
+	// Optional:
+	//  - HEAD-126: So we have the state paired with the bottom-most diff layer
+	//              if snapshot is enabled.
```

Comment:
Afaiu, you could just change `head-127` to `head-126`. I don't _think_ there's any specific reason as to why 127 is "Correct" and 126 would be "Wrong"

Reply:
I think I can revert the changes. Instead, we can maintain 128 diff layers in the momeory. So that we don't need to touch this part, don't need to flush in more data.